### PR TITLE
silx: init at 2.1.1, fabio: init at 24.4.0

### PR DESCRIPTION
**Pull request by pmiddend**
## Description of changes

Both fabio and silx are used at so-called synchroton radiation facilities all around the globe (particle accelerators that produce nice X-ray radiation). See [the fabio/silx website](http://www.silx.org/).

Originally, I wanted to package just silx, a viewer for experimental data in different formats, most notably HDF5, but it needed the fabio library, which is used by other tools as well.

I asked around and having this as one PR with two commits seems agreeable to people.

If you actually want to try running silx, download [a sample h5 file](https://zenodo.org/records/6497438/files/xrr_dataset.h5?download=1) and then after building silx, run `silx view xrr_dataset.h5`. It should show a tree of attributes on the left and actual data on the right.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by pmiddend** [comment]
@ofborg build silx

File: pkgs/by-name/si/silx/package.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    h5py
+    numpy
+    matplotlib
+    pyopengl
+    python-dateutil
+    pyside6
+    fabio
+  ];
+
+  meta = with lib; {
```
Comment (Aleksanaa) [inline pkgs/by-name/si/silx/package.nix:28]:
```suggestion
  meta = {
```

File: pkgs/development/python-modules/fabio/default.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  nativeBuildInputs = [
+    meson
+    python3Packages.cython
+    python3Packages.meson-python
+    ninja
+  ];
+
+  # While building, it tries to run version.py, which has a #!/usr/bin/env python3 shebang
+  postPatch = ''
+    sed -i 's#/usr/bin/env python3#${python3Packages.python}/bin/python3#' version.py
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/fabio/default.nix:31]:
```suggestion
    patchShebangs --build version.py
```

File: pkgs/development/python-modules/fabio/default.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  propagatedBuildInputs = with python3Packages; [
+    numpy
+    lxml
+    h5py
+    hdf5plugin
+    pillow
+    tomli
+  ];
+
+  meta = with lib; {
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/fabio/default.nix:43]:
```suggestion
  meta = {
```

File: pkgs/development/python-modules/fabio/default.nix [hunk 4]
```diff
@@ -0,0 +1,51 @@
+{
+  lib,
+  buildPythonPackage,
+  ninja,
+  fetchPypi,
+  meson,
+  python3Packages,
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/fabio/default.nix:7]:
```suggestion
```

You should not use this; the input is already in python3Packages (`python3xxPackages.callPackage`).

File: pkgs/by-name/si/silx/package.nix (hunk 1 above)
Comment (pmiddend) [reply pkgs/by-name/si/silx/package.nix]: Fixed this (twice) by making `lib.maintainers` and `lib.licenses` explicit. Hope that was the original issue.

File: pkgs/development/python-modules/fabio/default.nix (hunk 2 above)
Comment (pmiddend) [reply pkgs/development/python-modules/fabio/default.nix]: Fixed.

File: pkgs/development/python-modules/fabio/default.nix (hunk 3 above)
Comment (pmiddend) [reply pkgs/development/python-modules/fabio/default.nix]: Fixed

File: pkgs/development/python-modules/fabio/default.nix (hunk 4 above)
Comment (pmiddend) [reply pkgs/development/python-modules/fabio/default.nix]: Ah, didn't know about that. Fixed.

**Comment by pmiddend** [comment]
@Aleksanaa Thanks for the quick review!

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Comment by pmiddend** [comment]
@FliegendeWurst Thanks for the review! I've fixed all the issues (and reformatted with nixfmt).

File: pkgs/by-name/si/silx/package.nix [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  propagatedBuildInputs = with python3Packages; [
+    h5py
+    numpy
+    matplotlib
+    pyopengl
+    python-dateutil
+    pyside6
+    fabio
+  ];
+
```
Comment (FliegendeWurst) [inline pkgs/by-name/si/silx/package.nix:32]: Please make sure the tests are run.

File: pkgs/by-name/si/silx/package.nix [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    inherit pname version;
+    hash = "sha256-LfCRWkUrqQb7zxiFTPhy/g9FWhNMXTRbhEgek4tZb5I=";
+  };
+
+  nativeBuildInputs = with python3Packages; [
+    cython
+    setuptools
+  ];
+
+  propagatedBuildInputs = with python3Packages; [
```
Comment (FliegendeWurst) [inline pkgs/by-name/si/silx/package.nix:23]:
```suggestion
  dependencies = with python3Packages; [
```

File: pkgs/by-name/si/silx/package.nix [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  version = "2.1.1";
+
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-LfCRWkUrqQb7zxiFTPhy/g9FWhNMXTRbhEgek4tZb5I=";
+  };
+
+  nativeBuildInputs = with python3Packages; [
```
Comment (FliegendeWurst) [inline pkgs/by-name/si/silx/package.nix:18]:
```suggestion
  build-system = with python3Packages; [
```

File: pkgs/development/python-modules/fabio/default.nix [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+  dependencies = [
+    numpy
+    lxml
+    h5py
+    hdf5plugin
+    pillow
+    tomli
+  ];
+
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/fabio/default.nix:49]: Please make sure the tests are run.

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Comment by pmiddend** [comment]
@FliegendeWurst Tests have passed now. Is it ready to be merged?

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**